anitya/lib/model.py

```diff
@@ -362,15 +362,62 @@ def __json__(self, detailed=False):
         return output
 
     @classmethod
-    def get_or_create(cls, session, name, homepage, backend='custom'):
-        project = cls.by_name_and_homepage(session, name, homepage)
+    def _get_or_create(cls, session, name, homepage, ecosystem, backend,
+                       by='name_and_homepage'):
```

Comment:
I recommend adding a docblock to this method, at least to document the new ```by``` parameter and its possible values.